node_modules/gh-admins/lib/api.js

```diff
@@ -160,6 +161,58 @@ var updateGlobalAdmin = module.exports.updateGlobalAdmin = function(ctx, id, dis
 };
 
 /**
+ * Change a global administrator's password. The administator making the change will have to provide
+ * their own password. This is done to thwart situations where an administrator's session gets hijacked
+ *
+ * @param  {Context}        ctx                     Standard context containing the current user and the current app
+ * @param  {String}         id                      The id of the global administrator for which to change the password
+ * @param  {String}         newPassword             The new password for the global administrator
+ * @param  {String}         password                The password of the global administrator making the change
+ * @param  {Function}       callback                Standard callback function
+ * @param  {Object}         callback.err            An error that occurred, if any
+ */
+var changePassword = module.exports.changePassword = function(ctx, id, newPassword, password, callback) {
+    // Ensure the global administrator id is a valid number
+    id = GrasshopperUtil.getNumberParam(id);
+
+    // Validation
+    var validator = new Validator();
+    validator.check(null, {'code': 401, 'msg': 'Only global administrators can change the password of a global administrator'}).isGlobalAdmin(ctx);
+    validator.check(id, {'code': 400, 'msg': 'A valid global administrator id must be provided'}).isInt();
+    validator.check(newPassword, {'code': 400, 'msg': 'A valid new password must be provided'}).notEmpty();
+    validator.check(newPassword, {'code': 400, 'msg': 'A valid new password that is 6 characters or longer must be provided'}).len(6);
+    validator.check(password, {'code': 400, 'msg': 'A valid password must be provided'}).notEmpty();
+    if (validator.hasErrors()) {
+        return callback(validator.getFirstError());
+    }
+
+    // Ensure the global administrator that is making the change specified their own password
+    if (!AuthUtil.comparePassword(password, ctx.user.password)) {
+        return callback({'code': 401, 'msg': 'An incorrect password was provided'});
+    }
+
+    // Verify that the provided global administrator exists
+    getGlobalAdmin(ctx, id, function(err, globalAdmin) {
+        if (err) {
+            return callback(err);
+        }
+
+        AdminsDAO.changePassword(globalAdmin, newPassword, function(err) {
+            if (err) {
+                return callback(err);
+            }
+
+            log().info({
+                'actor': ctx.user,
+                'globalAdmin': globalAdmin
+            }, 'Updated the password of a global administrator');
```

Comment:
See https://github.com/fronteerio/grasshopper/blob/master/node_modules/gh-core/lib/db.js#L187 for how we do that